core/chains/evm/bulletprooftxmanager/eth_broadcaster.go

```diff
@@ -365,7 +364,7 @@ func (eb *EthBroadcaster) handleInProgressEthTx(etx EthTx, attempt EthTxAttempt,
 	if etx.State != EthTxInProgress {
 		return errors.Errorf("invariant violation: expected transaction %v to be in_progress, it was %s", etx.ID, etx.State)
 	}
-	parentCtx := context.TODO()
+	parentCtx := context.Background()
```

Comment:
I actually think this should be linked into a global struct context that is cancelled on Stop to allow early exiting.

Reply:
Sure, when we get to that point, this service will get the change. In this PR I migrated just one service and planned to migrate others in following PR(s). Or, I can do everything in this single PR, which will also remove these unpleasant constructs like `subservices []interface{} // services.Service or services.ServiceCtx`